### PR TITLE
compiler_rt: add addo

### PR DESCRIPTION
**Pull request by matu3ba**
- approach by Hacker's Delight with wrapping addition
- ca. 1.10x perf over the standard approach on my laptop
- tests with all combinations of min,max with -1,0,+1 and combinations of
  sequences +-1,2,4..,max

### CODE REVIEW

**Comment by matu3ba** [comment]
benchmark results: https://gist.github.com/matu3ba/f848765e3c67ee74781b57858049d692

see also #10248, which will be follow-up PRs.
If the intention is to not offer `addo` and `subo` to be used with external code for perf reasons, then this PR should be closed.

File: lib/std/special/compiler_rt/addosi4_test.zig [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    try test__addosi4(min, min);
+    try test__addosi4(max, max);
+    try test__addosi4(0, min);
+    try test__addosi4(0, max);
+    try test__addosi4(min, 0);
+    try test__addosi4(max, 0);
+    try test__addosi4(min, max);
+    try test__addosi4(max, min);
+
+    // zig fmt: off
```
Comment (g-w1) [inline lib/std/special/compiler_rt/addosi4_test.zig:59]: Why include these zig fmt directives? I think they are not necessary

Comment (matu3ba) [reply lib/std/special/compiler_rt/addosi4_test.zig:59]: zig fmt otherwise removes the alignment of the comments.

Comment (matu3ba) [reply lib/std/special/compiler_rt/addosi4_test.zig:59]: You are correct and I will remove them. I get confused with the behaviro of clang-fmt.

**Comment by andrewrk** [comment]
To be clear, #10248 has to do with the zig language; I believe this issue has no effect on the corresponding compiler-rt functions.

**Review by andrewrk — APPROVED** [review]

**Pull request merged**